src/allDrugs.jl

```diff
@@ -0,0 +1,68 @@
+""" In this file we fit all the drugs att once. """
+
+function getODEparamsAll(p::Array{Float64,1}, concentrations::Array{Float64,2})
+    effects = effects = zeros(9, 8, 4)
+
+    k = 1
+    # Scaled drug effect
+    for i=1:4
+        xx = 1.0 ./ (1.0 .+ (p[k] ./ concentrations[:, i]) .^ p[k+1])
+
+        effects[1, :, i] = p[k+2] .+ (p[k+3] - p[k+2]) .* xx
+        effects[2, :, i] = p[k+4] .+ (p[k+5] - p[k+4]) .* xx
+        effects[3, :, i] = p[k+6] .* xx
+        effects[4, :, i] = p[k+7] .* xx
+        effects[5, :, i] .= p[k+8]
+        k+=9
+    end
+    effects[6, :, :] .= floor(p[37]) #nG1
+    effects[7, :, :] .= floor(p[38]) #nG2
+    effects[8, :, :] .= floor(p[39]) #nD1
+    effects[9, :, :] .= floor(p[40]) #nD2
+
+    return effects
+end
+
+function residHillAll(hillParams::Array{Float64,1}, concentrations::Array{Float64,2}, g1::Array{Float64,3}, g2::Array{Float64,3})
+    res = Atomic{eltype(hillParams)}(0.0)
+    params = getODEparamsAll(hillParams, concentrations)
+
+    # Solve for all drugs
+    for j = 1:4
+        @threads for ii = 1:length(concentrations[:, j])
+            atomic_add!(
+                res,
+                cost(
+                    params[:, ii, j],
+                    g1[:, ii, j],
+                    g2[:, ii, j],
+                    Int(floor(params[6, ii, j])),
+                    Int(floor(params[7, ii, j])),
+                    Int(floor(params[8, ii, j])),
+                    Int(floor(params[9, ii, j])),
+                ),
+            )
+        end
+    end
+
+    return res[]
+end
+
+""" Hill optimization function for all drugs. """
+function optimize_hillAll(concs::Array{Float64,2}, g1::Array{Float64,3}, g2::Array{Float64,3}; maxstep = 1E5)
+    hillCostAll(hillParams) = residHillAll(hillParams, concs, g1, g2)
+
+    low = [minimum(concs[:, 1]), 0.01, 1e-9, 1e-9, 1e-9, 1e-9, 0.0, 0.0, 0.45, minimum(concs[:, 2]), 0.01, 1e-9, 1e-9, 1e-9, 1e-9, 0.0, 0.0, 0.45, minimum(concs[:, 3]), 0.01, 1e-9, 1e-9, 1e-9, 1e-9, 0.0, 0.0, 0.45, minimum(concs[:, 4]), 0.01, 1e-9, 1e-9, 1e-9, 1e-9, 0.0, 0.0, 0.45, 2, 10, 0, 0]
```

Comment:
Starting G1/G2 ratio can be shared. Control alpha and beta can be shared.

Reply:
@aarmey Is it alright to not do the second part of the tweak? I guess it doesn't reduce the number of parameters to be estimated, and based on this below, they are already being estimated closely.
![image](https://user-images.githubusercontent.com/34812750/75311091-9327d580-580a-11ea-9a56-0687ef4ded77.png)

Reply:
Sharing control alpha/beta doesn't reduce the number of parameters?

Reply:
I guess I'm wrong, the alpha and beta corresponding to control is just the minimum of alpha and beta in the hill, right? I thought they are just showing up after converting to the ODE parameters.